python/dgl/ops/sddmm.py

```diff
@@ -3,10 +3,32 @@
 import sys
 
 from ..backend import gsddmm as gsddmm_internal
+from ..backend import gsddmm_hetero as gsddmm_internal_hetero
 from .. import backend as F
 
 __all__ = ['gsddmm', 'copy_u', 'copy_v', 'copy_e']
 
+def reshape_lhs_rhs(lhs_data, rhs_data):
+    r""" Reshape the dimension of lhs and rhs data
+
+    Parameters
+    ----------
+    lhs_data : tensor or None
+        The left operand, could be None if it's not required by op.
+    rhs_data : tensor or None
+        The right operand, could be None if it's not required by op.
+    """
+    lhs_shape = F.shape(lhs_data)
+    rhs_shape = F.shape(rhs_data)
+    if len(lhs_shape) != len(rhs_shape):
+        max_ndims = max(len(lhs_shape), len(rhs_shape))
+        lhs_pad_ndims = max_ndims - len(lhs_shape)
+        rhs_pad_ndims = max_ndims - len(rhs_shape)
+        new_lhs_shape = (lhs_shape[0],) + (1,) * lhs_pad_ndims + lhs_shape[1:]
+        new_rhs_shape = (rhs_shape[0],) + (1,) * rhs_pad_ndims + rhs_shape[1:]
```

Comment:
can you give an example of the output shape? it seems you add a new dimension in the tensor, but the value of the added dimension isn't 1.

Reply:
Is this code copy-pasted from https://github.com/dmlc/dgl/blob/fc69f6cecad20335d076bf1c64cf03b2be5f6d37/python/dgl/ops/sddmm.py#L47-L60 ?

If so, please also call into this function inside the `if g._graph.number_of_etypes() == 1` branch of the `gsddmm` and `gspmm` functions to avoid duplication. Please also keep the original comment about the shape example per Da's suggestion.